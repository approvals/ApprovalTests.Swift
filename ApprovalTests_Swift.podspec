Pod::Spec.new do |s|
  s.name         = 'ApprovalTests_Swift'
  s.version      = '2.0.0'
  s.summary      = 'ApprovalTests for Swift, a powerful alternative to assertions'

  s.description = <<-DESC
                    Approval Tests are an alternative to assertions. Youâ€™ll find them useful for
                    testing objects with complex values (such as long strings), lots of properties,
                    or collections of objects.

                    ApprovalTests.Swift is compatible with the XCTest testing framework.
                  DESC

  s.homepage = 'https://github.com/approvals/ApprovalTests.Swift'
  s.license  = { :type => 'Apache License, Version 2.0', :file => 'LICENSE' }
  s.author   = { 'Jon Reid' => 'jon@qualitycoding.org' }
  s.social_media_url = 'https://twitter.com/qcoding'

  s.swift_version = '5.0'
  s.ios.deployment_target = '12.0'
  s.osx.deployment_target = '10.15'

  s.source = { :git => 'https://github.com/approvals/ApprovalTests.Swift.git', :tag => 'v2.0.0' }
  s.framework = 'XCTest'
  s.source_files      = "ApprovalTests.Swift/iOSApprovalsWatcher.py", "ApprovalTests.Swift/**/*.swift" 
  s.ios.exclude_files = "ApprovalTests.Swift/Reporters/macOS/*.swift"
  s.osx.exclude_files = "ApprovalTests.Swift/Reporters/iOS/*.swift"

end
