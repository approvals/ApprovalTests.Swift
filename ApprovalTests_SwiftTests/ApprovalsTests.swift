#if os(iOS)
  import ApprovalTests_iOS
#else
  import ApprovalTests_Swift
#endif
import XCTest
import Foundation

final class ApprovalsTests: XCTestCase {
    #if os(macOS)
        func testToString() throws {
            try Approvals.verify(CGRect(x: 5, y: 10, width: 100, height: 200))
        }
    #endif

    func testAsJson() throws {
        try Approvals.verifyAsJSON(CGRect(x: 5, y: 10, width: 100, height: 200))
    }

    func testAsJsonWithCustomObject() throws {
        try Approvals.verifyAsJSON(MyRect(x: 5, y: 10, width: 100, height: 200))
    }

    func testSequence() throws {
        var gameOfLife = GameOfLife { x, y in
            x >= 2 && x <= 4 && y == 2
        }
        try Approvals.verifySequence(gameOfLife, numberOfFrames: 2) { _ in
            gameOfLife = gameOfLife.advance
            return gameOfLife
        }
    }

  #if os(macOS)
    func testExecutableQuery() throws {
        try Approvals.verifyQuery(MetMuseumPieceLoader(45734))
    }
  #endif
}

class MetMuseumPieceLoader: ExecutableQuery {
    private let artID: Int

    init(_ artID: Int) throws {
        self.artID = artID
    }

    func getQuery() -> String {
        "https://collectionapi.metmuseum.org/public/collection/v1/objects/\(artID)"
    }

    func executeQuery(_ query: String) -> String {
        if query.isEmpty {
            return ""
        }
        return NetUtils.readURL(query).replacingOccurrences(matchingPattern: ",") { _ in ",\n" }
    }
}

private struct MyRect: Encodable {
    let x: Int
    let y: Int
    let width: Int
    let height: Int
}

private class GameOfLife {
    private var board: (Int, Int) -> Bool

    init(_ board: @escaping (Int, Int) -> Bool) {
        self.board = board
    }

    var advance: GameOfLife {
        GameOfLife { [self] x, y in
            let neighbors = [
                board(x - 1, y - 1),
                board(x - 0, y - 1),
                board(x + 1, y - 1),
                board(x - 1, y),
//              board(x - 0, y),
                board(x + 1, y),
                board(x - 1, y + 1),
                board(x - 0, y + 1),
                board(x + 1, y + 1),
            ]
            let cellCount = neighbors.filter { $0 }.count
            return cellCount == 3 || (cellCount == 2 && board(x, y))
        }
    }
}

extension GameOfLife: CustomStringConvertible {
    var description: String {
        StringUtils.printGrid(width: 5, height: 5) { x, y in
            board(x, y) ? "X " : ". "
        }
    }
}
