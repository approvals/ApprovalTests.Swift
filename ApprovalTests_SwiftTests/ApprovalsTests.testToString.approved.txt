(5.0, 10.0, 100.0, 200.0)
